src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -7586,7 +7586,8 @@ we will have to use something else.
                 }
                 Vec<ASR::expr_t*> args_; args_.reserve(al, x.n_args);
                 visit_expr_list(x.m_args, x.n_args, args_);
-                if (ASRUtils::is_array(ASRUtils::expr_type(args_[0])) &&
+                
+                if (x.n_args && ASRUtils::is_array(ASRUtils::expr_type(args_[0])) &&
```

Comment:
Is this change needed for this PR? What specific case does it fix?

If this change is not directly related to this PR, but was related to some issue found while working on this PR, then I suggest reporting the specific issue found and then sending a separate PR fixing it.

Reply:
It avoids an exception when the function is called with no parameters.

Reply:
>It avoids an exception when the function is called with no parameters.

Do you mean when the `type()` function is called with no paramaters?

Reply:
Yes. It is due to this check that we are able to check for `type()` called with no parameters in the `create_ObjectType()`. Without that check we have an unhandled exception.